[Event] Add method to request calendarView from Office365 API

krichprollsch opened the pull request:
Since the getlist do not return all recurring instances of an event,
calendarView gives a way to get the occurrences, exceptions, and single
instances of events

Taluu commented on src/Api/EventApi.php:
```diff
@@ -73,21 +100,57 @@ public function getList(Calendar $calendar = null, $filter = '', $orderBy = '')
+    public function getCalendarView(
+        Calendar $calendar = null,
+        DateTime $from = null,
+        DateTime $to = null,
```
these shouldn't be null, this is not the scope of this adapter to make things such as "magically pick an arbitrary date"

krichprollsch replied:
+1

krichprollsch replied:
fixed

Taluu commented on src/Api/EventApi.php:
```diff
@@ -56,7 +78,12 @@ public function getList(Calendar $calendar = null, $filter = '', $orderBy = '')
+                $calendar->getId(),
+                (new \DateTime('now'))->sub(new \DateInterval('P1Y'))->format('Y-m-d\TH:i:s\Z'),
+                (new \DateTime('now'))->add(new \DateInterval('P1Y'))->format('Y-m-d\TH:i:s\Z')
+            );
```
the `getList` shouldn't arbitrarly use the `calendarView` if there is a calendar

krichprollsch replied:
My bad, test I forgot to remove :/

krichprollsch replied:
fixed

krichprollsch committed:
[Event] Add method to request calendarView from Office365 API

Since the getlist do not return all recurring instances of an event,
calendarView gives a way to get the occurrences, exceptions, and single
instances of events

Taluu commented on src/Api/EventApi.php:
```diff
@@ -73,21 +94,54 @@ public function getList(Calendar $calendar = null, $filter = '', $orderBy = '')
+     * otherwise, it takes the primary
+     *
+     * If no from is given, it fetches one year before now
+     * If no to is given, it fetches one year after now
```
these comments are outdated

krichprollsch replied:
done

Taluu commented:
looks good. Please remove the comments about the date and it should be mergeable
